cpu/sam0_common/periph/uart.c

```diff
@@ -160,9 +160,22 @@ int uart_init(uart_t uart, uint32_t baudrate, uart_rx_cb_t rx_cb, void *arg)
     return UART_OK;
 }
 
+#ifdef MODULE_PERIPH_UART_NONBLOCKING
+void _flush_uart_tsrb(uart_t uart, size_t len) {
+    int byte = tsrb_get_one(&uart_tx_rb[uart]);
+    dev(uart)->DATA.reg = *byte;
```

Comment:
Should there be some waiting for the byte to actually be send? This seems to be done using the following in the rest of the implementation.

```C
while (!dev(uart)->INTFLAG.bit.DRE) {}
```